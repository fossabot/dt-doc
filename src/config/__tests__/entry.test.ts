import Config from 'webpack-chain';
import presetEntry from '../entry';

test('test preset entry', () => {
    const config = new Config();
    presetEntry(config);
    expect(config).toMatchSnapshot();
});
