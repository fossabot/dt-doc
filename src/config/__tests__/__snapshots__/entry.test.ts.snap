// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test preset entry 1`] = `
Object {
  "amd": [Function],
  "bail": [Function],
  "cache": [Function],
  "context": [Function],
  "devServer": Object {
    "after": [Function],
    "allowedHosts": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "before": [Function],
    "bonjour": [Function],
    "clientLogLevel": [Function],
    "color": [Function],
    "compress": [Function],
    "contentBase": [Function],
    "disableHostCheck": [Function],
    "filename": [Function],
    "headers": [Function],
    "historyApiFallback": [Function],
    "host": [Function],
    "hot": [Function],
    "hotOnly": [Function],
    "http2": [Function],
    "https": [Function],
    "index": [Function],
    "info": [Function],
    "inline": [Function],
    "lazy": [Function],
    "mimeTypes": [Function],
    "noInfo": [Function],
    "open": [Function],
    "openPage": [Function],
    "overlay": [Function],
    "parent": [Circular],
    "pfx": [Function],
    "pfxPassphrase": [Function],
    "port": [Function],
    "progress": [Function],
    "proxy": [Function],
    "public": [Function],
    "publicPath": [Function],
    "quiet": [Function],
    "setup": [Function],
    "shorthands": Array [
      "after",
      "before",
      "bonjour",
      "clientLogLevel",
      "color",
      "compress",
      "contentBase",
      "disableHostCheck",
      "filename",
      "headers",
      "historyApiFallback",
      "host",
      "hot",
      "hotOnly",
      "http2",
      "https",
      "index",
      "info",
      "inline",
      "lazy",
      "mimeTypes",
      "noInfo",
      "open",
      "openPage",
      "overlay",
      "pfx",
      "pfxPassphrase",
      "port",
      "proxy",
      "progress",
      "public",
      "publicPath",
      "quiet",
      "setup",
      "socket",
      "sockHost",
      "sockPath",
      "sockPort",
      "staticOptions",
      "stats",
      "stdin",
      "useLocalIp",
      "watchContentBase",
      "watchOptions",
      "writeToDisk",
    ],
    "sockHost": [Function],
    "sockPath": [Function],
    "sockPort": [Function],
    "socket": [Function],
    "staticOptions": [Function],
    "stats": [Function],
    "stdin": [Function],
    "store": Map {
      "stats" => "errors-only",
    },
    "useLocalIp": [Function],
    "watchContentBase": [Function],
    "watchOptions": [Function],
    "writeToDisk": [Function],
  },
  "devtool": [Function],
  "entryPoints": Object {
    "parent": [Circular],
    "store": Map {
      "entry" => Object {
        "parent": [Circular],
        "store": Set {
          ".doc\\\\entry.tsx",
        },
      },
    },
  },
  "externals": [Function],
  "loader": [Function],
  "mode": [Function],
  "module": Object {
    "defaultRules": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "noParse": [Function],
    "parent": [Circular],
    "rules": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "shorthands": Array [
      "noParse",
      "strictExportPresence",
    ],
    "store": Map {},
    "strictExportPresence": [Function],
  },
  "name": [Function],
  "node": Object {
    "parent": [Circular],
    "store": Map {},
  },
  "optimization": Object {
    "concatenateModules": [Function],
    "flagIncludedChunks": [Function],
    "mergeDuplicateChunks": [Function],
    "minimize": [Function],
    "minimizers": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "namedChunks": [Function],
    "namedModules": [Function],
    "noEmitOnErrors": [Function],
    "nodeEnv": [Function],
    "occurrenceOrder": [Function],
    "parent": [Circular],
    "portableRecords": [Function],
    "providedExports": [Function],
    "removeAvailableModules": [Function],
    "removeEmptyChunks": [Function],
    "runtimeChunk": [Function],
    "shorthands": Array [
      "concatenateModules",
      "flagIncludedChunks",
      "mergeDuplicateChunks",
      "minimize",
      "namedChunks",
      "namedModules",
      "nodeEnv",
      "noEmitOnErrors",
      "occurrenceOrder",
      "portableRecords",
      "providedExports",
      "removeAvailableModules",
      "removeEmptyChunks",
      "runtimeChunk",
      "sideEffects",
      "splitChunks",
      "usedExports",
    ],
    "sideEffects": [Function],
    "splitChunks": [Function],
    "store": Map {},
    "usedExports": [Function],
  },
  "output": Object {
    "auxiliaryComment": [Function],
    "chunkCallbackName": [Function],
    "chunkFilename": [Function],
    "chunkLoadTimeout": [Function],
    "crossOriginLoading": [Function],
    "devtoolFallbackModuleFilenameTemplate": [Function],
    "devtoolLineToLine": [Function],
    "devtoolModuleFilenameTemplate": [Function],
    "devtoolNamespace": [Function],
    "filename": [Function],
    "futureEmitAssets": [Function],
    "globalObject": [Function],
    "hashDigest": [Function],
    "hashDigestLength": [Function],
    "hashFunction": [Function],
    "hashSalt": [Function],
    "hotUpdateChunkFilename": [Function],
    "hotUpdateFunction": [Function],
    "hotUpdateMainFilename": [Function],
    "jsonpFunction": [Function],
    "library": [Function],
    "libraryExport": [Function],
    "libraryTarget": [Function],
    "parent": [Circular],
    "path": [Function],
    "pathinfo": [Function],
    "publicPath": [Function],
    "shorthands": Array [
      "auxiliaryComment",
      "chunkCallbackName",
      "chunkFilename",
      "chunkLoadTimeout",
      "crossOriginLoading",
      "devtoolFallbackModuleFilenameTemplate",
      "devtoolLineToLine",
      "devtoolModuleFilenameTemplate",
      "devtoolNamespace",
      "filename",
      "futureEmitAssets",
      "globalObject",
      "hashDigest",
      "hashDigestLength",
      "hashFunction",
      "hashSalt",
      "hotUpdateChunkFilename",
      "hotUpdateFunction",
      "hotUpdateMainFilename",
      "jsonpFunction",
      "library",
      "libraryExport",
      "libraryTarget",
      "path",
      "pathinfo",
      "publicPath",
      "sourceMapFilename",
      "sourcePrefix",
      "strictModuleExceptionHandling",
      "umdNamedDefine",
      "webassemblyModuleFilename",
    ],
    "sourceMapFilename": [Function],
    "sourcePrefix": [Function],
    "store": Map {},
    "strictModuleExceptionHandling": [Function],
    "umdNamedDefine": [Function],
    "webassemblyModuleFilename": [Function],
  },
  "parallelism": [Function],
  "parent": undefined,
  "performance": Object {
    "assetFilter": [Function],
    "hints": [Function],
    "maxAssetSize": [Function],
    "maxEntrypointSize": [Function],
    "parent": [Circular],
    "shorthands": Array [
      "assetFilter",
      "hints",
      "maxAssetSize",
      "maxEntrypointSize",
    ],
    "store": Map {},
  },
  "plugins": Object {
    "parent": [Circular],
    "store": Map {},
  },
  "profile": [Function],
  "recordsInputPath": [Function],
  "recordsOutputPath": [Function],
  "recordsPath": [Function],
  "resolve": Object {
    "alias": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "aliasFields": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "cachePredicate": [Function],
    "cacheWithContext": [Function],
    "concord": [Function],
    "descriptionFiles": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "enforceExtension": [Function],
    "enforceModuleExtension": [Function],
    "extensions": Object {
      "parent": [Circular],
      "store": Set {
        ".tsx",
        ".ts",
        ".js",
      },
    },
    "mainFields": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "mainFiles": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "modules": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "parent": [Circular],
    "plugins": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "shorthands": Array [
      "cachePredicate",
      "cacheWithContext",
      "concord",
      "enforceExtension",
      "enforceModuleExtension",
      "symlinks",
      "unsafeCache",
    ],
    "store": Map {},
    "symlinks": [Function],
    "unsafeCache": [Function],
  },
  "resolveLoader": Object {
    "alias": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "aliasFields": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "cachePredicate": [Function],
    "cacheWithContext": [Function],
    "concord": [Function],
    "descriptionFiles": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "enforceExtension": [Function],
    "enforceModuleExtension": [Function],
    "extensions": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "mainFields": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "mainFiles": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "moduleExtensions": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "modules": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "packageMains": Object {
      "parent": [Circular],
      "store": Set {},
    },
    "parent": [Circular],
    "plugins": Object {
      "parent": [Circular],
      "store": Map {},
    },
    "shorthands": Array [
      "cachePredicate",
      "cacheWithContext",
      "concord",
      "enforceExtension",
      "enforceModuleExtension",
      "symlinks",
      "unsafeCache",
    ],
    "store": Map {},
    "symlinks": [Function],
    "unsafeCache": [Function],
  },
  "shorthands": Array [
    "amd",
    "bail",
    "cache",
    "context",
    "devtool",
    "externals",
    "loader",
    "mode",
    "name",
    "parallelism",
    "profile",
    "recordsInputPath",
    "recordsPath",
    "recordsOutputPath",
    "stats",
    "target",
    "watch",
    "watchOptions",
  ],
  "stats": [Function],
  "store": Map {},
  "target": [Function],
  "watch": [Function],
  "watchOptions": [Function],
}
`;
